services/mongodb/mongodb.go

```diff
@@ -0,0 +1,137 @@
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+* details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* version 3 along with this program in the file "LICENSE".  If not, see
+* <http://www.gnu.org/licenses/agpl-3.0.txt>.
+*
+* See https://honeytrap.io/ for more details. All requests should be sent to
+* licensing@honeytrap.io
+*
+* The interactive user interfaces in modified source and object code versions
+* of this program must display Appropriate Legal Notices, as required under
+* Section 5 of the GNU Affero General Public License version 3.
+*
+* In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+* these Appropriate Legal Notices must retain the display of the "Powered by
+* Honeytrap" logo and retain the original copyright notice. If the display of the
+* logo is not reasonably feasible for technical reasons, the Appropriate Legal Notices
+* must display the words "Powered by Honeytrap" and retain the original copyright notice.
+ */
+package mongodb
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net"
+
+	"github.com/honeytrap/honeytrap/event"
+	"github.com/honeytrap/honeytrap/pushers"
+	"github.com/honeytrap/honeytrap/services"
+	logging "github.com/op/go-logging"
+)
+
+/*
+
+[service.mongodb]
+type="mongodb"
+version="2.5.1"
+dbs=[ {Name="My_DB", SizeOnDisk="8192", Empty="false"},
+      {Name="example", SizeOnDisk="32768", Empty="false"}, ]
+
+[[port]]
+port="tcp/27017"
+services=["mongodb"]
+
+*/
+
+var log = logging.MustGetLogger("services/mongodb")
+
+var (
+	_ = services.Register("mongodb", Mongodb)
+)
+
+func Mongodb(options ...services.ServicerFunc) services.Servicer {
+	s := &mongodbService{
+		mongodbServiceConfig: mongodbServiceConfig{
+			Version: "3.6.3",
+			Dbs: []Db{
+				{"admin", "32768", "false"},
+				{"config", "12288", "false"},
+				{"local", "122880", "false"},
+			},
+		},
+	}
+	for _, o := range options {
+		o(s)
+	}
+	return s
+}
+
+type mongodbServiceConfig struct {
+	Version string `toml:"version"`
+	Dbs     `toml:"dbs"`
+}
+
+type Db struct {
+	Name       string
+	SizeOnDisk string
+	Empty      string
+}
+
+type Dbs []Db
+
+type mongodbService struct {
+	mongodbServiceConfig
+	ch pushers.Channel
+}
+
+func (s *mongodbService) SetChannel(c pushers.Channel) {
+	s.ch = c
+}
+
+func (s *mongodbService) Handle(ctx context.Context, conn net.Conn) error {
+
+	defer conn.Close()
+
+	br := bufio.NewReader(conn)
+	port := conn.RemoteAddr().(*net.TCPAddr).Port
+
+	for {
+		buff := make([]byte, 1024)
+		n, err := br.Read(buff[:])
+		if err == io.EOF {
+			return nil
+		} else if err != nil {
+			log.Error("Bad request: %s", err.Error())
+		}
+		b := bytes.NewBuffer(buff[:n])
+
+		answer := s.MongoDBHandler(port, b)
+
+		s.ch.Send(event.New(
```

Comment:
Could you extract specific data in connection request, like username, password, application etc.